src/compiler/gcc.rs

```diff
@@ -364,62 +362,26 @@ pub fn compile<T>(creator: &T,
         }
     };
 
-    // When reading from stdin the language argument is needed
+    // Pass the language explicitly as we might have gotten it from the
+    // command line.
     let language = match parsed_args.language {
-        Language::C => "cpp-output",
-        Language::Cxx => "c++-cpp-output",
-        Language::ObjectiveC => "objective-c-cpp-output",
-        Language::ObjectiveCxx => "objective-c++-cpp-output",
+        Language::C => "c",
+        Language::Cxx => "c++",
+        Language::ObjectiveC => "objective-c",
+        Language::ObjectiveCxx => "objective-c++",
     };
     let mut attempt = creator.clone().new_command_sync(executable);
```

Comment:
nit: we could probably rename this something other than `attempt` now that it's the only command.